app/models/assignment.rb

```diff
@@ -55,6 +56,8 @@ class Assignment < ApplicationRecord
 
   attr_accessor :reasons_for_rejection
 
+  before_save :mark_case_as_dirty_for_responding_assignments
```

Comment:
I originally used before_save here because I wanted access to the fields that had changed, but I see you can get them in after_save as well.  There are issues with after commit - see https://apidock.com/rails/ActiveRecord/Transactions/ClassMethods/after_commit, although these bugs may have been addressed by now.  What do you think the implications of saving a case before the assignment might be?  I am reluctant to change it just because  - well, because what?   Am happy to change if there's a real issue, but not otherwise.

Reply:
It's just a smell for me ... what if something prevents the save from occuring by either stopping the chain or raising? What if we remove the 10 second delay and re-indexing stops working because of the race condition? If you think these concerns are too marginal to worry about fine, let's just roll with it as-is.

Reply:
Write a ticket to get it converted to after save or after commit if it bothers you that much, and convince Polly why we should spend time on it.  My view is that it doesn't really matter what order they're done in, as long as they all get done, or none get done.  After save has to be used if you're writing a new associated record and need the id of that record,  but in this case it isn't, and it seems to me better to make sure the record is marked as dirty before writiing the assignment.